sycl/include/CL/sycl/detail/pi.hpp

```diff
@@ -99,9 +99,9 @@ using PiMemObjectType = ::pi_mem_type;
 using PiMemImageChannelOrder = ::pi_image_channel_order;
 using PiMemImageChannelType = ::pi_image_channel_type;
 
-void contextSetExtendedDeleter(const cl::sycl::context &constext,
-                               pi_context_extended_deleter func,
-                               void *user_data);
+__SYCL_EXPORT void contextSetExtendedDeleter(const cl::sycl::context &constext,
```

Comment:
I don't see any users of it. Why does this need to be exported from libsycl?

Reply:
Please also document the recently added PI API: piextContextSetExtendedDeleter in pi.h

Reply:
Why is it not implemented for OpenCL?

Reply:
Hi, this function is used in the cuBLAS backend to OneMKL. If it isn't exported, the cuBLAS backend does not work.

https://github.com/oneapi-src/oneMKL/pull/2#discussion_r409005148

Good idea, the function will be documented in subsequent commits.

It has no equivelant in OpenCL backend because it isn't required by any OpenCL driven projects. Is there a way you would like this to be marked as CUDA specific?

Reply:
I think using any symbols from `sycl::detail` namespace is dangerous as it's by design intended for internal implementation details. 
I suggest oneMKL to use "public/standard/stable" API if possible.

tagging @mkrainiuk, @mehdi-goli.

Reply:
This is meant to be a temporary fix until there is a SYCL extension to cover for this functionality, that's why its exposing some implementation details, because there is no public API for this at this point.

Reply:
Is there a SYCL extension in flight to support this?

What's exactly the functionality achieved here (it's hard to tell for sure because there is not documentation of this extension)? 

I'd like to approve this to unblock MKL, but please create an issue (or just PR) to document the `piextContextSetExtendedDeleter` and also to discuss if there is a better way to achieve what's needed.